src/auto/list_box.rs

```diff
@@ -183,10 +185,30 @@ pub trait ListBoxExt: 'static {
 }
 
 impl<O: IsA<ListBox>> ListBoxExt for O {
-    //#[cfg(any(feature = "v3_16", feature = "dox"))]
-    //fn bind_model<P: Fn(&glib::Object) -> Widget + 'static, Q: Into<Option<P>>>(&self, model: /*Ignored*/Option<&gio::ListModel>, create_widget_func: Q) {
-    //    unsafe { TODO: call ffi::gtk_list_box_bind_model() }
-    //}
+    #[cfg(any(feature = "v3_16", feature = "dox"))]
+    fn bind_model<'a, P: IsA<gio::ListModel> + 'a, Q: Into<Option<&'a P>>, R: Fn(&glib::Object) -> Widget + 'static, S: Into<Option<R>>>(&self, model: Q, create_widget_func: S) {
+        let model = model.into();
+        let create_widget_func = create_widget_func.into();
+        let create_widget_func_data: Box_<Option<R>> = Box::new(create_widget_func.into());
+        unsafe extern "C" fn create_widget_func_func<'a, P: IsA<gio::ListModel> + 'a, Q: Into<Option<&'a P>>, R: Fn(&glib::Object) -> Widget + 'static>(item: glib_ffi::gpointer, user_data: glib_ffi::gpointer) -> *mut ffi::GtkWidget {
+            let callback: &Option<R> = &*(user_data as *mut _);
+            let res = if let Some(ref callback) = *callback {
+                callback()
+            } else {
+                panic!("cannot get closure...")
+            };
+            res.to_glib()
+        }
+        let create_widget_func = if create_widget_func_data.is_some() { Some(create_widget_func_func::<'a, P, Q, R> as _) } else { None };
```

Comment:
There's a mismatch with the pointer type of the first argument here. `*mut libc::c_void` vs. `*mut gobject_ffi::GObject`. The sys crate is correct, the code generated here is wrong.